### DISCUSSION TITLE
Find if an md code is a valid mermaid syntax and which diagram it is

### DISCUSSION THREAD

**Discussion by YossiSaadi** [mermaid-js/mermaid#4154] [Q&A]
Hello :)
I saw on the source code that there's a method named `detectType` that returns the diagram type of an MD string.
Was wondering, as it's not exposed, which ways I have to figure if something is a Mermaid diagram and which diagram it is.

My use case is getting an MD document and finding if it contains any diagrams.

**Comment by sidharthv96** [comment]
detectType will not detect mermaid diagram inside markdown documents directly. You'll have to pare md document and then filter out code blocks with type `mermaid`. If you pass the string inside the code block, then `detectType` can figure out which diagram type it is. 

We could export detectType in a future release if it's a requirement.

**Reply by YossiSaadi** [reply]
Thanks @sidharthv96!
Yep, I'm aware of that, sorry for not being specific.
I meant supplying the method with an MD that I suspect contains Mermaid diagram, not an entire document.

Would love to see this being exported, thanks! 💪🏼

**Comment by YossiSaadi** [comment]
@sidharthv96 Thank you, just saw it was exported!
